linera-core/src/client.rs

```diff
@@ -93,6 +95,7 @@ impl<ValidatorNodeProvider: Clone> ChainClientBuilder<ValidatorNodeProvider> {
         validator_node_provider: ValidatorNodeProvider,
         max_pending_messages: usize,
         cross_chain_message_delivery: CrossChainMessageDelivery,
+        tracked_chains: HashSet<ChainId>,
```

Comment:
very optional nit: Maybe take an `impl Iterator` to remove a couple of `HashSet::from(...)`

Reply:
Good idea, done!